src/parser/jobs/vpr/modules/SerpentsTail.tsx

```diff
@@ -54,50 +69,120 @@ export class SerpentsTail extends Analyser {
+			this.firstLegaciesGained++
+		} else if (event.action === this.data.actions.SECOND_GENERATION.id) {
+			this.secondLegaciesGained++
+		} else if (event.action === this.data.actions.THIRD_GENERATION.id) {
+			this.thirdLegaciesGained++
+		} else if (event.action === this.data.actions.FOURTH_GENERATION.id) {
+			this.fourthLegaciesGained++
+		}
 	}
 
-	private onTailSpenders() {
-		this.tailsDone++
+	private legacySpent(event: Events['action']) {
+		if (event.action === this.data.actions.FIRST_LEGACY.id) {
+			this.firstLegaciesUsed++
+		} else if (event.action === this.data.actions.SECOND_LEGACY.id) {
+			this.secondLegaciesUsed++
+		} else if (event.action === this.data.actions.THIRD_LEGACY.id) {
+			this.thirdLegaciesUsed++
+		} else if (event.action === this.data.actions.FOURTH_LEGACY.id) {
+			this.fourthLegaciesUsed++
+		}
 	}
 
 	private onComplete() {
+
+		const Counters = {
+			deathRattle: {action: this.data.actions.DEATH_RATTLE, ready: this.deathRattleReadies, done: this.deathRattles},
+			lastLash: {action: this.data.actions.LAST_LASH, ready: this.LashingReadies, done: this.LastLashings},
+			firstLegacy: {action: this.data.actions.FIRST_LEGACY, ready: this.firstLegaciesGained, done: this.firstLegaciesUsed},
+			secondLegacy: {action: this.data.actions.SECOND_LEGACY, ready: this.secondLegaciesGained, done: this.secondLegaciesUsed},
+			thirdLegacy: {action: this.data.actions.THIRD_LEGACY, ready: this.thirdLegaciesGained, done: this.thirdLegaciesUsed},
+			fourthLegacy: {action: this.data.actions.FOURTH_LEGACY, ready: this.fourthLegaciesGained, done: this.fourthLegaciesUsed},
+		}
+
 		this.checklist.add(new Rule({
-			name: <Trans id = "VPR.serpentstail.waste.name"> Use your <DataLink action="SERPENTS_TAIL"/> follow-ups </Trans>,
+			name: <Trans id="VPR.serpentstail.waste.name"> Use your <DataLink action="SERPENTS_TAIL"/> follow-ups </Trans>,
 			displayOrder: DISPLAY_ORDER.SERPENTS_TAIL,
 			target: PERFECTIO,
 			description: <Trans id="vpr.serpentstail.waste.content">
-				Using <DataLink action = "HINDSBANE_FANG"/>, <DataLink action="HINDSTING_STRIKE"/>, <DataLink action="FLANKSBANE_FANG"/> or <DataLink action="FLANKSTING_STRIKE"/> grant <DataLink action="DEATH_RATTLE"/>.
+				Using <DataLink action="HINDSBANE_FANG"/>, <DataLink action="HINDSTING_STRIKE"/>, <DataLink action="FLANKSBANE_FANG"/> or <DataLink action="FLANKSTING_STRIKE"/> grants <DataLink action="DEATH_RATTLE"/>.
 				<br/>
-				Using <DataLink action = "JAGGED_MAW"/> and <DataLink action="BLOODIED_MAW"/> grant <DataLink action="LAST_LASH"/>.
+				Using <DataLink action="JAGGED_MAW"/> and <DataLink action="BLOODIED_MAW"/> grants <DataLink action="LAST_LASH"/>.
 				<br/>
 				Using the Generation skills under <DataLink action="REAWAKEN"/> grants Legacy follow up skills.
 				<br/>
 				These skills are important to a Viper's damage and must be used immediately after the skill that grants them by using <DataLink action="SERPENTS_TAIL">.</DataLink>
```

Comment:
How weird that this DataLink for Serpent's Tail is only showing the icon, and not the text...

Reply:
Fixed. having a . between the datalink and /datalink 100% broke it.